src/metadata.rs

```diff
@@ -94,6 +94,18 @@ impl Metadata {
     pub fn links(&self) -> Option<&String> {
         self.links.as_ref()
     }
+
+    /// Yank this version
+    pub fn yank(&mut self) {
+        // FIXME: What if it's already yanked?
+        self.yanked = true;
+    }
```

Comment:
i think not checking the state is ok for `Metadata`, it's basically just a setter.